jwql/instrument_monitors/common_monitors/cosmic_ray_monitor.py

```diff
@@ -0,0 +1,608 @@
+        convention
+
+    ValueError
+        If the most recent query search returns more than one entry
+    """
+
+    def __init__(self):
+        """Initialize an instance of the ``Cosmic_Ray`` class."""
+
+    def identify_tables(self):
+        """Determine which database tables to use for a run of the
+        cosmic ray monitor.
+
+        Uses the instrument variable to get the mixed-case instrument
+        name, and uses that name to find the query and stats tables
+        for that instrument.
+        """
+
+        mixed_case_name = JWST_INSTRUMENT_NAMES_MIXEDCASE[self.instrument]
+        self.query_table = eval('{}CosmicRayQueryHistory'.format(mixed_case_name))
+        self.stats_table = eval('{}CosmicRayStats'.format(mixed_case_name))
+
+    def get_cr_mags(self, jump_locs, jump_locs_pre, rateints, jump_data, jump_head):
+        """Gets the magnitude of each cosmic ray.
+
+        Computes a list of magnitudes using the coordinate of the
+        detected jump compared to the magnitude of the same pixel in
+        the group prior to the jump.
+
+        Parameters:
+        ----------
+
+        jump_locs: list
+            List of coordinates to a pixel marked with a jump.
+
+        jump_locs_pre: list
+            List of matching coordinates one group before jump_locs.
+
+        rateints: ndarray
+            Array in DN/s.
+
+        jump_data: ndarray
+            Ndarray containing image data cube
+
+        jump_head: FITS Header
+            FITS header unit containing information about the jump data
+
+        Returns:
+        -------
+
+        mags: list
+            A list of cosmic ray magnitudes corresponding to each jump.
+
+        """
+
+        mags = []
+        for coord, coord_gb in zip(jump_locs, jump_locs_pre):
+            mags.append(self.magnitude(coord, coord_gb, rateints, jump_data, jump_head))
+
+        return mags
+        
+
+    def get_jump_data(self, jump_filename):
+        """Opens and reads a given .FITS file containing cosmic rays.
+
+        Parameters:
+        ----------
+        jump_filename: str
+            Path to file.
+
+        Returns:
+        -------
+        head: FITS header
+            Header containing file information
+
+        data: NoneType
+            FITS data
+
+        dq: ndarray
+            Data Quality array containing jump flags.
+
+        """
+
+        hdu = fits.open(jump_filename)
+
+        head = hdu[0].header
+        data = hdu[1].data
+
+        dq = hdu[3].data
+
+        hdu.close()
+
+        return head, data, dq
+
+    def get_jump_locs(self, dq):
+        """Uses the data quality array to find the location of all
+        jumps in the data.
+
+        Parameters:
+        ----------
+        dq: ndarray
+            Data Quality array containing jump flags.
+
+        Returns:
+        -------
+        jump_locs: list
+            List of coordinates to a pixel marked with a jump.
+        """
+
+        temp = np.where(dq == 4)
```

Comment:
Are you explicitly looking for groups flagged as containing a jump and nothing else? With this line as-is, if you have a pixel that contains any other flag and then has a jump flag added on top, it won't be found.

If you do want to catch jumps in pixels that contain other flags, I usually use this:
from jwst.datamodels import dqflags
jump = np.where(dq & dqflags.pixel["JUMP_DET"] > 0)